### ISSUE TITLE
FITS table -> Astropy.table does not strip spaces

### ISSUE THREAD

**Issue by stscieisenhamer** [astropy/astropy#2608]
### System info

OS X 10.9.3
Python 2.7.5
Astropy 0.4.dev11
### Issue

In converting a FITS Table to an Astropy Table, if a column, of type string, has values that are right-padded with spaces, the spaces are not stripped.

Below is an example. Note that using the FITS_rec directly, the trailing spaces do not appear. However, after conversion, they do.

```
Environment setup
In [5]:
    from astropy.io import fits
    from astropy.table import Table

In [2]:
    hdulist = fits.open('x6q17587l_phot.fits')

In [3]:
    hdulist[1].data['OPT_ELEM']

Out[3]:
chararray(['G130M', 'G130M', 'G130M', 'G130M', 'G130M', 'G130M', 'G130M',
       'G130M', 'G130M', 'G130M', 'G160M', 'G160M', 'G160M', 'G160M',
       'G160M', 'G160M', 'G160M', 'G160M', 'G160M', 'G160M', 'G140L',
       'G140L', 'G140L', 'G140L', 'G130M', 'G130M', 'G130M', 'G130M',
       'G130M', 'G130M', 'G130M', 'G130M', 'G130M', 'G130M', 'G160M',
       'G160M', 'G160M', 'G160M', 'G160M', 'G160M', 'G160M', 'G160M',
       'G160M', 'G160M', 'G140L', 'G140L', 'G140L', 'G140L', 'G140L',
       'G140L', 'G140L', 'G140L', 'G130M', 'G130M', 'G130M', 'G130M',
       'G130M', 'G130M', 'G130M', 'G130M', 'G130M', 'G130M', 'G130M',
       'G130M'], 
      dtype='|S8')

In [6]:
    data_table = Table(hdulist[1].data)

In [7]:
    data_table['OPT_ELEM']

Out[7]:
<Column name='OPT_ELEM' unit=None format=None description=None>
array(['G130M   ', 'G130M   ', 'G130M   ', 'G130M   ', 'G130M   ',
       'G130M   ', 'G130M   ', 'G130M   ', 'G130M   ', 'G130M   ',
       'G160M   ', 'G160M   ', 'G160M   ', 'G160M   ', 'G160M   ',
       'G160M   ', 'G160M   ', 'G160M   ', 'G160M   ', 'G160M   ',
       'G140L   ', 'G140L   ', 'G140L   ', 'G140L   ', 'G130M   ',
       'G130M   ', 'G130M   ', 'G130M   ', 'G130M   ', 'G130M   ',
       'G130M   ', 'G130M   ', 'G130M   ', 'G130M   ', 'G160M   ',
       'G160M   ', 'G160M   ', 'G160M   ', 'G160M   ', 'G160M   ',
       'G160M   ', 'G160M   ', 'G160M   ', 'G160M   ', 'G140L   ',
       'G140L   ', 'G140L   ', 'G140L   ', 'G140L   ', 'G140L   ',
       'G140L   ', 'G140L   ', 'G130M   ', 'G130M   ', 'G130M   ',
       'G130M   ', 'G130M   ', 'G130M   ', 'G130M   ', 'G130M   ',
       'G130M   ', 'G130M   ', 'G130M   ', 'G130M   '], 
      dtype='|S8')
```

**Comment by pllim** [comment]
I just ran into this. Would be nice it to have a solution!

**Comment by divenex** [comment]
Like many users of AstroPy Table, I've also spent time debugging why my two string columns weren't matching.

I realized that you must explicitly strip the whitespaces. As this decade-old AstroPy issue might never be resolved, here's an updated solution, for those dealing with this bug, using the new, more efficient [numpy.strings.strip](https://numpy.org/doc/stable/reference/generated/numpy.strings.strip.html) function from NumPy 2.

````
from astropy.table import Table
import numpy as np

t = Table.read('filename.fits')
for name in t.colnames: 
    if t[name].dtype.char == 'S':
        t[name] = np.strings.strip(t[name])
```

**Comment by taldcroft** [comment]
Would it be worthwhile to add this as a convenience option for reading FITS via `Table`? I.e. `Table.read(filename, strip=True)`?

I'm actually in the same camp that trailing whitespace in FITS files has caused me a ton of headaches through the years. Even in the last week this bit me. It's somewhat locked into the API, sadly, and we already get feedback about introducing too many breaking changes.

**Comment by neutrinoceros** [comment]
I'm in favor of adding this flag, though I'd suggest a more univocal name like `strip_whitespace`.

**Comment by taldcroft** [comment]
@neutrinoceros - agreed and fixed. :smile:

**Comment by saimn** [comment]
I started something for that last time this issue was active but got lost in the pile of other issues:
https://github.com/saimn/astropy/commit/25b2628c414694b01a6e4f7c17dda3de93f08d7a

`chararray.rstrip()` uses `np.strings` I think so perf should be the same.
Adding an option to trigger this might be useful for people not wanted to pay the performance price, but if not activated by default will people find it easily ? (Not sure people know/think at using `Table.read.help`)

**Comment by taldcroft** [comment]
> but if not activated by default will people find it easily ? (Not sure people know/think at using Table.read.help)

Agreed about people not finding this anyway. I asked two people in my team that are reasonably experienced with astropy Table how they would find the available options for reading a FITS file with Table. Neither of them got to `Table.read.help` right away. Googling was the first answer, and that sends people to the `io.fits` package docs, which is basically not helpful.

So it's really the wrong place, but I've thinking about this problem independently and wondering if the Unified I/O layer is causing more problems than it solves. What about being more like other packages and having independent methods for each format. So `Table.read_fits(..)` or `QTable.read_ecsv()`.

**Comment by mhvk** [comment]
@saimn - the `np.strings` stuff is new in numpy 2.0, and uses a proper ufunc, so performance should be better than the python loop that used to be there.
